File: app/src/main/java/com/marvelcomics/brito/marvelcomics/ui/fragment/character/CharacterFragment.java
package com.marvelcomics.brito.marvelcomics.ui.fragment.character;


import android.content.Context;
import android.databinding.BindingAdapter;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.marvelcomics.brito.entity.CharacterEntity;
import com.marvelcomics.brito.infrastructure.utils.MarvelThumbnailAspectRatio;
import com.marvelcomics.brito.marvelcomics.R;
import com.marvelcomics.brito.marvelcomics.databinding.FragmentCharacterBinding;
import com.squareup.picasso.Picasso;

public class CharacterFragment extends Fragment {

    public static final String ARGUMENT_CHARACTER = "character_args";

    private CharacterEntity characterEntity;
    private FragmentCharacterBinding binding;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        Bundle args = getArguments();
        characterEntity = args.getParcelable(ARGUMENT_CHARACTER);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        binding = DataBindingUtil.inflate(inflater, R.layout.fragment_character, container, false);
        binding.setCharacter(characterEntity);
        return binding.getRoot();
    }

    @BindingAdapter("android:src")
    public static void setCharacterImage(ImageView imageView, CharacterEntity character) {
        Context context = imageView.getContext();
        Picasso.with(imageView.getContext())
                .load(character.getFullUrlThumbnailWithAspect(MarvelThumbnailAspectRatio.Portrait.XLARGE))
                .placeholder(context.getDrawable(R.drawable.progress))
                .into(imageView);
    }

    public static CharacterFragment newInstance(CharacterEntity characterEntity) {
        CharacterFragment fragment = new CharacterFragment();

        Bundle args = new Bundle();
        args.putParcelable(ARGUMENT_CHARACTER, characterEntity);

        fragment.setArguments(args);
        return fragment;
    }
}

File: app/src/main/java/com/marvelcomics/brito/view/fragment/character/CharacterFragment.kt
package com.marvelcomics.brito.view.fragment.character

import android.os.Bundle
import android.view.View
import androidx.fragment.app.Fragment
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions
import com.marvelcomics.brito.R
import com.marvelcomics.brito.databinding.FragmentCharacterBinding
import com.marvelcomics.brito.domain.entity.CharacterEntity
import com.marvelcomics.brito.infrastructure.utils.MarvelThumbnailAspectRatio
import com.marvelcomics.brito.view.extensions.viewBinding

class CharacterFragment : Fragment(R.layout.fragment_character) {

    private val binding by viewBinding(FragmentCharacterBinding::bind)
    private var characterEntity: CharacterEntity? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val args = arguments
        characterEntity = args?.getSerializable(ARGUMENT_CHARACTER) as CharacterEntity
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        binding.textviewFragmentCharacterName.text = characterEntity?.name
        binding.textviewFragmentCharacterDescription.text = characterEntity?.description
        Glide.with(this)
            .load(
                characterEntity?.getFullUrlThumbnailWithAspect(
                    MarvelThumbnailAspectRatio.Portrait.MEDIUM
                )
            )
            .apply(RequestOptions().centerCrop())
            .into(binding.imageviewFragmentCharacter)
    }

    companion object {
        const val ARGUMENT_CHARACTER = "character_args"

        fun newInstance(characterEntity: CharacterEntity): CharacterFragment {
            val fragment = CharacterFragment()
            val args = Bundle()
            args.putSerializable(ARGUMENT_CHARACTER, characterEntity)
            fragment.arguments = args
            return fragment
        }
    }
}
